[wanja] - 헤더 새로고침 및 시간 표시 기능 추가 / import 정렬 + alias 설정 / 버튼·뱃지·모달 공용 컴포넌트 생성/ 피드백 반영 / grid view 컴포넌트 , list view 컴포넌트 / 뉴스 롤링 기능

wan0514 opened the pull request:
## 작업 완료 내역

### ✨ 기능 추가
- **헤더 로고 클릭 시 페이지 새로고침 및 현재 시간 표시 기능 구현**
  - 헤더 로고 클릭 시 페이지가 새로고침되도록 설정
  - `YYYY.MM.DD (요일)` 형식으로 현재 시간 표시
  - 날짜 포맷 유틸 함수 생성
- **공용 Badge 컴포넌트 추가 및 구독 뱃지 대체**
  - `isActive` 여부에 따라 조건부 렌더링 지원
  - `MediaTab` 전용 구독 뱃지를 제거하고 공용 Badge 컴포넌트로 대체
- **구독 해지 모달 컴포넌트 구현**
  - 사용자가 **구독 해지 버튼**을 눌렀을 때 나타나는 **모달 UI 컴포넌트**를 구현했습니다.
  - 해지 대상인 `media` 이름을 **props로 전달받아**, 모달 내에 **동적으로 노출**되도록 처리했습니다.
- **Grid View 컴포넌트 구현**  
  - 테이블 기반 레이아웃 구현 
  - 미디어 항목을 일정한 테이블 구조로 배치하여 정돈된 UI 구성
- **GridView, ListView 컴포넌트 목데이터로 동적 렌더링 처리**  
  - `data` prop을 통해 전달받은 데이터를 기반으로 각 미디어 항목을 동적으로 렌더링함
  - `View`가 `grid / list` 여부에 따라 전달하는 데이터를 다르게 함.(아래 고민과 해결과정에 상세 설명)

---
#### ✨ 4/3 목요일 구현 사항
- **뉴스 롤링 기능 구현**
  - 세로 방향 **롤링 뉴스 UI 컴포넌트** 동적 랜더링
  - 일정 간격마다 자동으로 다음 뉴스로 슬라이드됨
  - **hover 시 롤링 일시 정지**, hover 해제 시 재개 기능 포함
     - `setInterval` → **재귀 `setTimeout` 구조로 전환**하여  슬라이드 타이밍 꼬임 없이 안정적인 애니메이션 구현


---

### 🐛 버그 수정
- **토글 버튼 동작 오류 수정**
  - 잘못 전달된 함수 prop으로 인한 버튼 미작동 문제 해결
- **그리드 아이콘 렌더링 오류 수정**
  - `iconMap`의 키 오타(`gird` → `grid`) 수정으로 렌더링 문제 해결

---

### 💄 스타일 설정
- **ESLint import 정렬 규칙 적용 및 전체 import 재정렬**
- **경로 alias 적용 및 관련 ESLint 규칙 업데이트**
  - 간결한 import 작성을 위해 alias 도입
  - alias 기반 정렬을 위해 ESLint 설정 수정
  - import 그룹 간 간격 비활성화로 tighter block 구성
- **import 경로 및 컴포넌트 이름 오타 수정**
  - alias 구조에 맞도록 경로 수정
  - 컴포넌트 이름 오타 수정

---

### 🏗️ 구조 및 설계
- **StackLayout 컴포넌트 생성 및 세로 간격 제어 기능 도입**
  - 전체 레이아웃을 관리하는 `StackLayout` 컴포넌트 추가
  - 자식 컴포넌트 간의 세로 간격을 prop 기반으로 제어 가능
- **기존 Layout 컴포넌트 리팩토링**
  - `nth-child` 기반 마진 제어 → 동적 세로 간격 제어 방식으로 전환
- **MediaTab 컴포넌트 분리**
  - `TabList`, `ViewModeToggle` 컴포넌트로 분리하여 관심사 분리(Separation of Concerns) 강화


## 주요 고민 및 해결

### 1. **컴포넌트 간 간격을 어떻게 유연하게 제어할 것인가?**

#### 🧐 고민의 배경

여러 개의 컴포넌트를 `App` 컴포넌트에서 조립하는 과정에서, 각 컴포넌트 간 간격이 서로 달라야 하는 요구사항이 있었습니다.

- 모든 컴포넌트 간 간격이 동일하다면 `flex + gap` 조합으로 간단히 해결 가능
- 하지만 이번 디자인은 **컴포넌트마다 간격이 다르기 때문에 개별 제어가 필요**

#### 고려한 방식

1. **각 컴포넌트 내부에서 prop으로 간격 값을 받아 처리**
   - ✅ *장점*: 별도의 래퍼 컴포넌트 없이 간단히 구현 가능  
   - ❌ *단점*: 사용하는 곳마다 마진 값을 명시해야 하므로 반복적이고 번거로움

2. **레이아웃 전용 컴포넌트(`StackLayout`)를 만들어 자식 간 간격을 제어**
   - ✅ *장점*: 레이아웃 관련 로직을 한 곳에서 관리할 수 있어 유지보수 용이  
   - ❌ *단점*: 컴포넌트 트리에 래퍼가 추가됨

→ **결론적으로 2번 방식을 선택**, 레이아웃을 한 곳에서 관리하는 구조가 유지보수에 유리하다고 판단함.

그 후 2번 방식 안에서도 두 가지 구현 방식을 검토함:

- **A. Emotion의 `:nth-child` 선택자를 활용한 스타일 제어**
  - ✅ 코드가 간결하고 선언적
  - ❌ 자식 컴포넌트 개수나 순서가 바뀌면 관리가 어려움
  - ❌ `spacing` 값을 동적으로 조정하기 어려움
  - ❌ 조건부 렌더링 시 마진 순서가 꼬일 수 있음

- **B. `children.map`을 사용하여 자식마다 동적으로 마진 적용**
  - ✅ 자식 개수가 바뀌어도 유연하게 대응 가능
  - ✅ `spacing` 배열 등을 활용해 간격을 컴포넌트 외부에서 제어 가능
  - ❌ 다소 장황한 코드

#### ✅ 최종 해결

**`children.map`을 이용해 동적으로 마진을 제어하는 방식**을 최종 선택했습니다.

이 방식은 다음과 같은 장점이 있습니다:

- 자식 수나 순서가 바뀌어도 안전하게 동작
- 간격을 외부에서 유연하게 조절 가능
- 조건부 렌더링에도 대응 가능
- 추후 다른 곳의 레이아웃 용으로 재사용이 가능함!

> 초기에 `:nth-child` 방식의 간결함이 매력적이라 고려했지만,  
> 추후 간격을 조정해야할 요소가 늘어나거나, 순서가 바뀔 경우 nth-child 코드를 계속 작성해줘야하기 때문에 제어하기 편한 방식은 map방식이라고 판단했습니다.
> 또한 레이아웃 간격 조절용으로 다른 곳에서 쓰인다면 동적 방식이 적절하다 판단했습니다.

--- 

### 2. **점점 길어지는 import 경로, 어떻게 정리할 수 있을까?**

#### 🧐 고민의 배경

컴포넌트를 세분화하면서 상대 경로(`../../../`)가 길어지고 복잡해졌습니다.  
또한 파일마다 import 순서가 달라 가독성이 떨어졌고, 이를 **일관된 정렬 기준과 짧은 경로**로 정리하고자 했습니다.

목표는 다음과 같았습니다:

- `@/components/...` 형태의 **alias 경로 사용**
- alias 경로도 포함하여 **import 정렬 자동화**

#### 고려한 방식

- **Alias 설정**
  - `vite.config.js`에서 `@`, `@components` 등의 경로를 설정
  - ES Module 환경의 `__dirname` 문제는 `fileURLToPath` + `path.dirname`으로 해결

- **ESLint import 정렬**
  - `eslint-plugin-import`, `eslint-import-resolver-alias` 사용
  - `@emotion/**`는 `external`, `@/**`는 `internal`로 명시
  - 정렬 순서는 `builtin > external > internal > parent/sibling/index` 기준
  - `newlines-between` 옵션으로 그룹 간 줄바꿈 제어

#### ✅ 최종 해결

- alias 경로(`@components/...` , `@/...` )를 적용해 가독성 향상  
- import 정렬 설정을 통해 **모듈 분류 + 자동 정렬** 모두 적용  
- `@emotion/**`를 `external` 그룹으로 먼저 지정해  분리(❗해결되지 않은 문제❗)

#### ❗ 해결되지 않은 이슈

- 여전히 `@emotion/styled`가 `react`와 다른 그룹으로 묶이며 **불필요한 줄바꿈** 발생  
- 원하는 정렬 방식은 아래와 같지만, 완벽하게 반영되지 않음:

  ```
  1. builtin
  2. external
  3. internal (@components 등)
  4. 상대 경로 (../, ./)
  ```

- 디버깅 도구(`DEBUG=eslint-plugin-import:*`)를 통해 분류 과정을 확인했으나,  
  일부 패키지가 의도한 그룹으로 분류되지 않아 **임시로 그룹 간 줄바꿈을 비활성화**하고 보류 상태로 남겨두었습니다.😢

--- 

### 3. **Badge를 어디서, 어떻게 붙일 것인가?**

#### 🧐 고민의 배경

- 전체 언론사 / 구독 언론사 탭은 구조가 같아 **공통 컴포넌트(TabItem)** 로 재사용 중이었음  
- 초기에 `TabItem`, `TabItem`, `Badge`를 같은 레벨에서 나란히 배치하고, margin으로 정렬함  
- 하지만 **Badge는 구독 탭과 연관된 정보**이므로 **TabItem 안으로 포함되어야 맞다**고 판단함  
- 결국 Badge를 TabItem 내부에 조건부 렌더링해야 하는 구조가 되었고, **그 방식**에 대해 고민함  

#### 고려한 방식

| 방식 | 설명 | 장점 | 단점 |
| --- | --- | --- | --- |
| 1. `badgeLabel` prop 방식 | Badge 유무를 prop으로 제어 | 직관적, 재사용성 높음 | TabItem이 Badge를 알아야 함 |
| 2. `label`에 JSX 전달 | `<label + Badge />` 조합 전달 | 관심사 분리, 유연함 | 직관성 떨어짐 |
| 3. `children`으로 전달 | TabItem 내용을 직접 구성 | 가장 유연함 | 너무 많은 책임이 사용자(개발자)에게 있음 |

#### ✅ 최종 해결

- JSX 조합 방식은 `<>{text}<Badge /></>`처럼 생소하고 직관적이지 않아 **가독성에서 아쉬움**
- 구조적으로도 **Badge를 조건으로 붙이는 단순한 상황**이라, 고정된 조합이 더 적합

👉 따라서  
`label="텍스트"`, `badgeLabel="8"`처럼 **명시적인 prop 방식**을 선택함

이는 공용 컴포넌트에 적합하며, 컴포넌트의 **책임 분리**, **유지보수**, **확장성** 면에서 모두 장점이 있음.

---

### 4. 아이콘 컴포넌트를 어떻게 렌더링할 것인가?

#### 🧐 고민의 배경

- 버튼 컴포넌트 내에서 아이콘을 `iconType`에 따라 분기 렌더링해야 하는 상황
- JSX 안에 삼항 연산자로 바로 렌더링하면 가독성이 떨어짐

```jsx
{icon === 'plus' ? <PlusIcon /> : <ClosedIcon />}
```

- 아이콘 크기 제어 및 재사용성도 고려하여, 아이콘 렌더링 방식 자체를 어떻게 할지 설계하고자 함.


#### 고려한 방식

1. JSX 내에서 함수 실행  
   - 예: {getIconComponent(iconType)}  
   - 장점: 간단, 함수화 가능  
   - 단점: JSX 가독성 떨어짐, 디버깅 어려움

2. 변수로 저장 후 사용  
   - 예: const IconComponent = iconType === 'plus' ? <PlusIcon /> : <ClosedIcon />  
   - 장점: JSX 구조 깔끔, 조건 처리 쉬움  
   - 단점: 삼항 연산이 많아지면 복잡

3. iconMap을 외부에 선언  
   - 예: const IconComponent = iconMap[iconType]  
   - 장점: 구조화 쉬움, 확장성 좋음  
   - 단점: 선언 위치와 사용 위치가 분리됨

4. iconMap을 함수 내부에 넣고 변수로 저장  

```jsx
     function getIconComponent(type) {
       const map = { plus: PlusIcon, closed: ClosedIcon };
       return map[type] || null;
     }  
     const IconComponent = getIconComponent(iconType);
```
   - 장점: 가독성, 조건 처리, 디버깅, 확장성 모두 좋음  
   - 단점: 코드 한 줄 늘어남

#### ✅ 최종 선택

- iconMap을 함수 내부에 선언하고, JSX에서는 변수를 통해 컴포넌트를 렌더링하는 방식 선택
> 이 방법이 추후 iconComponent에 조건부를 걸거나, fallback등을 사용하거나 등의 수정이 필요할 경우 가장 유연한 구조라 판단하였고,
> 또한, return 문의 구조가 확실하게 보여서 가독성에 좋기 때문에, 코드가 한 줄 늘어나는 단점은 문제가 되지 않다 생각했습니다.

```jsx
function getIconComponent(type) {
  const map = {
    plus: PlusIcon,
    closed: ClosedIcon,
  };
  return map[type] || null;
}

const IconComponent = getIconComponent(iconType);

return (
  <IconWrapper>
    <IconComponent />
  </IconWrapper>
);
```


### 4.📆 날짜 포맷팅 유틸 함수 리팩토링과 `useMemo` 활용 결정 과정

## 1. 최초 구현 방식

처음에는 날짜 포맷팅 함수(`getFormattedToday`)를 `new Date()`를 함수 내에서 생성하여 오늘 날짜와 요일을 반환하는 형태로 구현했습니다. 간편하게 사용할 수 있다는 장점이 있었습니다.

```javascript
const getFormattedToday = () => {
  const now = new Date();
  // 날짜 + 요일 포맷 반환
};
```

## 2. 구조 개선

유틸 함수의 재사용성을 높이기 위해 `new Date()`를 함수 외부에서 받도록 개선했습니다. 이 방식은 테스트 가능성과 재사용성을 증가시켰습니다.

```javascript
const getFormattedToday = (date) => {
  // 전달받은 date 객체를 기반으로 날짜 포맷 처리
};
```

## 3. 성능에 대한 고민

그러나 이 구조에서는 컴포넌트 리렌더링 시마다 `new Date()`가 생성되고, 포맷팅 함수가 반복적으로 호출되는 성능 낭비가 발생할 수 있겠다는 생각이 들었습니다. 만약 날짜 포맷 형식이 시간을 포함했다면 현재 방식을 유지해도 되지만, 날짜와 요일만 표기되며 이는 24시간 안에서는 변하지 않으므로 리랜더링 될때마다 같은 계산을 할 필요가 없다 판단되어 이를 캐싱할 수 있는 방법을 찾아보았습니다.

## 4. 해결 방법 – `useMemo` 도입

이 문제를 해결하기 위해 `useMemo`를 도입하여 날짜 정보는 최초 렌더링 시 한 번만 계산되도록 최적화했습니다.

```jsx
const today = useMemo(() => {
  const now = new Date();
  return getFormattedToday(now);
}, []); // 의존성 없음 → 최초 1회만 계산
```

### ✅ `useMemo`를 사용한 이유:
- 리렌더링 시 반복 계산을 방지하고, 최초 렌더링에서 한 번만 계산하도록 최적화했습니다.
- `getFormattedToday` 함수는 유연하게 작성되어 `useMemo`와 잘 결합되었습니다.

---

### 5. **Border가 겹치는 문제**

#### 🧐 고민  
컴포넌트 간의 **border 이중 겹침 현상**으로 인해 시각적으로 두꺼운 테두리가 보이는 문제가 있었습니다. 이를 해결하기 위해 다양한 방법을 검토했습니다:

- **`table` 태그 + `border-collapse`**  
  border 겹침을 방지하는 데 효과적이지만, 단순 스타일링 목적으로 table을 사용하는 것은 **시멘틱 구조상 부적절**하다고 판단했습니다.
  
- **오른쪽 요소에만 border 적용**  
  이 방법은 중복 border를 피할 수 있지만, 동일한 `Button` 컴포넌트를 반복 사용하는 구조였기 때문에  
  `:nth-child` 등을 활용해 **컴포넌트 내부에서 border 방향을 조건부로 지정해야 했고**,  
  이는 코드가 **복잡하고 직관성이 떨어진다**고 느꼈습니다.

#### ✅ 최종 해결  
결국 아래 방식으로 간단하고 유지보수하기 쉬운 해결책을 선택했습니다:

- **부모 요소에 `display: flex`와 `gap: 1px`을 지정**
- gap으로 생긴 공간의 **배경색을 border 색상과 동일하게 설정**

이를 통해 시각적으로는 border가 있는 것처럼 보이면서도,  
**겹침 없이 자연스럽고 명확한 구분선**을 만들 수 있었습니다.

---

### 6. **모달이 열릴 때 배경 스크롤 방지**

#### 🧐 고민  
모달이 열릴 때, 배경의 스크롤이 가능한 상태를 방지하고자 했습니다.  
단순히 `overflow: hidden`을 설정했을 때, 모달이 닫힌 후에도 스크롤이 **복원되지 않는 문제**가 발생했습니다.

#### ✅ 해결  
- 모달이 열릴 때:  
  `document.body.style.overflow = 'hidden'`을 통해 스크롤을 차단했습니다.
- 모달이 닫힐 때:  
  `useEffect`에서 **모달의 열림 상태**(`isModalOpen`)를 의존성 배열에 포함시켜,  
  상태가 변경될 때마다 `overflow` 값을 **정확히 복원**하도록 처리했습니다.
  ```jsx
  useEffect(() => {
    if (isModalOpen) {
      document.body.style.overflow = 'hidden';
    } else {
      document.body.style.overflow = '';
    }
  }, [isModalOpen]);
  ```

---

### 7. **그리드 뷰와 리스트 뷰의 데이터를 어떻게 효율적으로 전달할 것인가?**

#### 1. 🧐 고민의 배경

- **Grid View**는 언론사 `id`, `로고 이미지`, `구독 여부` 정도의 간단한 정보만 필요합니다.  
- 반면 **List View**는 카테고리별로 약 90개의 언론사 데이터를 요구하며, 각 언론사는 다음과 같은 상세한 정보를 포함합니다:
  - 언론사 id, 이름, 로고, 시간
  - 메인 뉴스 (id, 타이틀, 이미지, 링크)
  - 서브 뉴스들 (각각 id, 타이틀, 링크)

➡️ **즉, View Type에 따라 요구되는 데이터의 구조와 양이 전혀 다릅니다.**  
➡️ 따라서 각 View에 맞는 데이터를 따로 fetch하고, 필요한 구조로 나누어 전달하는 것이 효율적이라고 판단했습니다.


#### 2. 😥 처음 시도했던 방식과 문제점

처음에는 `GridView`, `ListView`가 필요로 하는 데이터 구조가 다르기 때문에, 각각의 데이터를 개별 prop으로 넘겨 자식 컴포넌트 내부에서 조건 분기로 골라 쓰는 방식도 고려했습니다.

```jsx
return <ViewComponent listData={listData} gridData={gridData} />;
```

하지만 이 방식은 여러 문제를 동반했습니다:

- ❌ **불필요한 데이터가 함께 전달되므로 비효율적**
- ❌ **컴포넌트 내부에서 viewType에 따라 조건 분기를 해야 하므로 로직이 지저분해짐**
- ❌ **prop 개수가 늘어나면서 인터페이스가 복잡해짐**
- ❌ 결과적으로 데이터를 분리한 의미가 사라지고, **컴포넌트 책임 분리가 흐려짐**


#### 3. ✅ 개선 방향: `data` prop 하나로 통일

하위 컴포넌트에는 `data`라는 prop 하나만 전달하고, **어떤 데이터를 넘길지는 상위에서 `viewType`에 따라 결정**하는 방식으로 구조를 변경했습니다.

```jsx
<MediaSection
  viewType={viewType}
  data={viewType === 'grid' ? gridData : listData}
/>
```

`MediaSection` 내부에서도 `viewType`에 따라 알맞은 컴포넌트를 선택해 렌더링하고, 동일한 prop 이름으로 데이터를 전달합니다:

```jsx
const ViewComponent = getViewComponent(viewType);
return <ViewComponent data={data} />;
```

```jsx
function getViewComponent(type) {
  return {
    grid: GridView,
    list: ListView,
  }[type];
}
```

> ✅ 결과적으로 각 View 컴포넌트는 자신에게 필요한 데이터만 받아 처리하고,  
> 상위에서는 `viewType`에 따라 데이터를 선택적으로 넘겨주는 방식으로  
> **컴포넌트 간 역할 분리와 prop 인터페이스 단순화**를 모두 달성했습니다.


#### 4. 🤔 fetch는 어디서 해야 할까?

이 과정에서 또 하나 고민한 포인트는:  
**"데이터를 어디서 fetch해야 할까?"** 였습니다.

처음에는 `MediaSection` 내부에서 `useEffect`를 통해 데이터를 직접 가져오는 것도 고려했습니다:

```jsx
useEffect(() => {
  fetch('/mockData/pressList.json').then(...);
}, []);
```

하지만 이 방식은 다음과 같은 이유로 적절하지 않다고 판단했습니다:

- 📌 **역할이 모호해짐** – `MediaSection`은 UI 컴포넌트로서 렌더링에만 집중해야 함
- 📌 **재사용성이 떨어짐** – 외부에서 데이터를 공급하거나 캐싱하기 어려움
- 📌 **로딩/에러 처리의 분산** – 결국 상태를 다시 상위로 끌어올려야 하므로 구조만 복잡해짐
- 📌 **디버깅/테스트 어려움** – 내부 fetch는 mocking이나 제어가 까다로움


#### 5. ✅ 최종 구조

따라서 데이터 fetch는 **상위 컴포넌트(`Media`)에서 담당**하고,  
**`MediaSection`에는 단순히 `viewType`과 `data`만 전달**하는 구조로 정리했습니다.

```jsx
const [gridData, setGridData] = useState([]);
const [listData, setListData] = useState([]);

useEffect(() => {
  fetch('/mockData/pressList.json')
    .then((res) => res.json())
    .then(setGridData)
    .catch((err) => console.error('gridData fetch error:', err));

  fetch('/mockData/pressNews.json')
    .then((res) => res.json())
    .then(setListData)
    .catch((err) => console.error('listData fetch error:', err));
}, []);
```

```jsx
<MediaSection
  viewType={viewType}
  data={viewType === 'grid' ? gridData : listData}
/>
```
---

### 8. **슬라이드 타이밍 꼬임 현상 해결**  ( 4/ 3 목요일 )

#### 🤔  고민의 배경
세로 롤링 뉴스 UI에서 `setInterval`과 `setTimeout`을 조합해 슬라이드를 제어하던 구조에서,  
**hover 후 재생 시 슬라이드 타이밍이 꼬이거나 어색한 애니메이션이 발생**하는 문제를 발견했습니다.

#### 📌 문제 원인
- `setInterval`이 **슬라이드 완료 여부와 무관하게** 주기적으로 실행됨  
- `hover` 중단 후에도 `setTimeout`이 예약된 상태로 실행되며 **중복 발생**  
- 이로 인해 **UX가 어색**하고 **제어 어려움**

#### ✅ 해결 방법
- `setInterval` 제거
- **재귀 `setTimeout` 구조로 전환**하여  
  슬라이드가 **완료된 직후에만 다음 슬라이드를 예약**하도록 개선

```js
function scheduleNextSlide() {
  timeoutRef.current = setTimeout(() => {
    setIsSliding(true);

    setTimeout(() => {
      setIsSliding(false);
      setIndex((prev) => prev + 1);
      scheduleNextSlide(); // 🔁 다음 슬라이드 예약
    }, 500);
  }, 5000);
}
```

#### 🧠 기대 효과
- 슬라이드 타이밍이 **정확하게 유지**
- `hover` 중단 후에도 **중복 예약 없이 안정적 동작**
- 구조가 단순해져 **유지보수성 향상**

wan0514 committed:
♻️ refactor: create StackLayout component with vertical spacing control

- Added StackLayout component to manage overall layout
- Layout component handles vertical margin between child components

wan0514 committed:
♻️ refactor: update existing Layout component to use dynamic vertical spacing

- Replaced nth-child based margin logic with dynamic vertical spacing
- Layout component now adjusts spacing between children programmatically

wan0514 committed:
♻️ refactor: split MediaTab into TabList and ViewModeToggle components

- Extracted TabList and ViewModeToggle from MediaTab for better separation of concerns
- Created individual subcomponents to improve modularity and maintainability

wan0514 committed:
💄 style : Apply ESLint import sort rule and reorder all imports

wan0514 committed:
💄 style : Apply file alias and update ESLint import rules accordingly

- Introduced file path alias for cleaner and shorter imports
- Updated ESLint import rules to support alias-based sorting
- Disabled import group spacing option for tighter import blocks

wan0514 committed:
🐛 fix : Fix invisible grid icon due to typo in iconMap key

- Corrected typo from 'gird' to 'grid' in iconMap reference
- Fixed issue where grid icon was not rendering on UI

wan0514 committed:
🐛 fix : Fix button toggle bug caused by incorrect function prop

- Resolved issue where toggle button was unresponsive
- Fixed improper function prop passing that prevented toggle behavior

wan0514 committed:
💄 style : Fix import paths and correct component name typos

- Updated incorrect import paths to match alias structure
- Fixed typos in component names to ensure proper rendering and usage

wan0514 committed:
✨ feat : Add header logo refresh and current time display

- Implemented page refresh on header logo click
- Displayed current time in header using YYYY.MM.DD (Day) format
- Created utility function to format date as specified

wan0514 committed:
✨ feat: add common Badge component with isActive conditional rendering

- Created reusable Badge component with support for isActive-based conditional rendering
- Replaced MediaTap-specific subscription badge with the new common Badge component
- Removed the old MediaTap-only SubscribeBadge component

wan0514 committed:
✨ feat: add common Button component with dynamic styles

- Create common Button component for reuse across project
- Support isEmphasized, label existence, iconType, and hover state
- Style dynamically adjusts based on props provided

wan0514 committed:
🐛 fix: correct invalid border color value in Button component

- Changed border.bold token value for clearer visual hierarchy
- Previously, border.bold and border.default shared the same color

crongro commented:
> * 자식 수나 순서가 바뀌어도 안전하게 동작
> * 간격을 외부에서 유연하게 조절 가능
> * 조건부 렌더링에도 대응 가능
> * 추후 다른 곳의 레이아웃 용으로 재사용이 가능함!
> 
> >

과정이 무척 좋네요. 
유지보수를 고려하기가 어려운데 그런점을 충분히 고려한거 같아요.

crongro commented:
> ```js
> function getIconComponent(type) {
>   const map = {
>     plus: PlusIcon,
>     closed: ClosedIcon,
>   };
>   return map[type] || null;
> }
> ```

깔끔하네요. ㅎ

crongro commented on eslint.config.js:
```diff
@@ -31,6 +33,48 @@ export default [
+            {
+              pattern: '@emotion/**',
+              group: 'external',
+              position: 'after',
```
이거 제거해도 줄바꿈이 생겨요?

wan0514 replied:
> 이거 제거해도 줄바꿈이 생겨요?

...해결되었습니다 감사합니다 😭😭😭😭
path group하면 같은 external 그룹이어도 하위 그룹이 생겨서 무조건 간격이 띄워지는지 몰랐습니다...

crongro commented on src/components/Media/MediaTab/ViewModeToggle/index.jsx:
```diff
@@ -0,0 +1,26 @@
+      <ViewButton
+        type="list"
+        isActive={isListView}
+        onClick={() => setIsListView(true)}
```
지금처럼 해도 좋고요, 
좀더 직관적이려면, onClick={() => setViewMode('list')} 
이런식으로 표현해도 좋을거 같네요.

wan0514 replied:
> 지금처럼 해도 좋고요, 좀더 직관적이려면, onClick={() => setViewMode('list')} 이런식으로 표현해도 좋을거 같네요.

타입이 두 경우 뿐이라 toggle처럼 boolean 방식을 택했는데
확실히 개발하다 보니 type을 string으로 받는 것이 더 가독성이 좋다는 걸 느끼고 있습니다... 감사합니다!

crongro commented on src/utils/getFormattedToday.js:
```diff
@@ -0,0 +1,20 @@
+export default function getFormattedToday() {
```
유틸이면 범용적으로 동작하면 더 좋으니 date와 같은것을 매개변수로 받아도 좋죠.

crongro reviewed (approved):
꽤 클린해요.
그러니 구현방식에 대해서 너무 깊게 고민하진마시고 
기능을 완성해보세요.

wan0514 commented:
> 꽤 클린해요. 그러니 구현방식에 대해서 너무 깊게 고민하진마시고 기능을 완성해보세요.

균형 잡기가 아직 어려운 거 같습니다 🥲
이번주에는 기능 구현 속도 내는 걸 우선적으로 해보겠습니다!

wan0514 committed:
♻️ refactor: enhance date formatting utility for reusability and performance

- Modified getFormattedToday to accept a date parameter for better flexibility
- Moved Date instantiation to the header component for clearer responsibility
- Used useMemo to cache formatted date and improve performance on re-renders

wan0514 committed:
Merge pull request #1 from wan0514/refactor/date-formatter

♻️ refactor: 날짜 포맷 함수 변경, useMemo사용하여 성능 높이기

wan0514 committed:
Merge pull request #2 from wan0514/feature/button-component

Feat: button component
Fixed: border style bugs

wan0514 committed:
♻️ refactor: replace boolean view and tab states with string values for clarity

- Changed tab and view type states from boolean to string for better readability
- Improved clarity when passing props and handling logic in child components

wan0514 committed:
🐛 fix: resolve line break issue with emotion imports in external group

- Removed @emotion/** from pathGroups in eslint-plugin-import config
- Prevented unintended line breaks between react and emotion imports
- Ensured emotion is treated as external and sorted right after react

wan0514 committed:
✨ feat: add alert modal with isModalOpen and targetItem state

- Added isModalOpen state to control the visibility of the modal
- Passed targetItem to the modal to display specific item information

wan0514 committed:
♻️ refactor: rename Alert component and related variables to Modal

- Renamed Alert component to Modal for consistency
- Updated all related variable and prop names from alert to modal

wan0514 committed:
🐛 fix: restore accidentally deleted media section

- Recovered the media area that was mistakenly removed
- Ensured all related components and styles are intact

crongro commented:
> 이번주에는 기능 구현 속도 내는 걸 우선적으로 해보겠습니다!

그리고 기술정리 내용이 좋은데, 노션같은 곳에 일지처럼 옮겨 담는것도 생각해보세요.

wan0514 committed:
✨ feat: create grid view UI for dynamic media list rendering

- Implemented grid-based layout to display media items
- mediaList prop is used to render media data dynamically

wan0514 committed:
🍰 sample: remove sample button for testing modal open/close

wan0514 committed:
✨ feat: Design ListView component UI

- Dynamic data not connected yet

wan0514 committed:
✨ feat : render Media Section by prop data

fetch mockdata yet

wan0514 committed:
✨ feat: implement dynamic data rendering with fetch for list/grid view

- Created mock data for use in both list and grid views
- Used fetch to dynamically render content based on the selected view
- List view currently displays a single news agency as a test
- Plan to fetch full agency list and implement swapping-based rendering later

wan0514 committed:
♻️ refactor: split MainNewsItem and SubNewsItem components in ListView

- Extracted MainNewsItem and SubNewsItem into separate components
- Improved code readability and modularity within the ListView component

wan0514 committed:
🙈 chore: remove unnecessary files

wan0514 committed:
✨ feat: implement dynamic rendering for news rolling area with mock data

- Created mock data for news rolling section and fetched it dynamically
- Split data into chunks of 5 items and passed to individual rolling items
- Used useMemo to optimize rendering performance
- Currently only renders the first item of each chunk as rolling logic is not yet implemented

crongro commented on src/components/Header/DateDisplay/index.jsx:
```diff
@@ -9,7 +10,12 @@ const DateDisplay = styled.div`
 
 const Header = () => {
-  return <DateDisplay>{getFormattedToday()}</DateDisplay>;
+  const today = useMemo(() => {
```
memoization 의 장점이 있긴하죠. 이런 새로운 훅 사용은 좋고요. 
최적화 훅이니 성능차이가 날만한 곳에서 사용하면 더 좋겠죠.

crongro commented on src/components/Media/index.jsx:
```diff
@@ -11,59 +11,24 @@ const MediaContainer = styled.div`
   const [viewType, setViewType] = useState('grid');
 
+  useEffect(() => {
+    fetch('/mockData/pressList.json') // grid view용 데이터
+      .then((res) => res.json())
+      .then(setGridData)
+      .catch((err) => console.error('gridData fetch error:', err));
+
+    fetch('/mockData/pressNews.json') // list view용 데이터
+      .then((res) => res.json())
+      .then(setListData)
+      .catch((err) => console.error('listData fetch error:', err));
```
로딩시점말고, 필요한 시점에 데이터를 가져올 수도 있으니, 그런 방법과도 장단점을 생각해보세요.

wan0514 committed:
✨ feat: implement vertical rolling animation for news rolling section

- Added vertical rolling functionality to the news rolling area
- Prepared structure for dynamic switching between rolling items

wan0514 committed:
✨ feat: pause/resume rolling on hover and style title on hover

- Implemented pause and resume functionality for vertical rolling on mouse hover
- Added underline to the title and changed cursor to pointer on hover

wan0514 committed:
🐛 fix: resolve React key warning by using index as map key

- Fixed warning caused by duplicate keys in components rendered via map
- Issue occurred due to copying the first data item to the end while retaining the same id
- Replaced duplicated id-based key with map index to ensure uniqueness

wan0514 committed:
♻️ refactor: simplify rolling logic to avoid key issues

- Initially tried to create seamless rolling by duplicating the first of 5 items at the end
- Faced key duplication issue due to reused id, temporarily resolved by using map index
- This prevented passing meaningful id data to components and reduced flexibility
- Refactored approach to handle only 2 items, improving clarity and scalability

wan0514 committed:
🍰 sample: Generate mock data from crawled dataset

- Create mock data based on the structure of the crawled news data
- Facilitate front-end development and UI testing using realistic sample entries

wan0514 committed:
✨feat: render grid and list components based on new mock data

- Rendered grid and list components using the new mock data
- Changed list view to show only the first media of the selected category on click
- Auto-switching between media and swipe features not implemented yet
- Current logic is for testing purposes only

wan0514 committed:
💄 style: add highlight color and count label for selected category

- Applied distinct color styling for selected category
- Added count label next to each category to indicate number of items

wan0514 committed:
💄 style: update grid hover and truncate list titles

- On grid view, changed cell background and hid logo on hover
- In list view, applied CSS to truncate long sub-article titles with ellipsis

crongro commented on croller/parser.cjs:
```diff
@@ -0,0 +1,59 @@
+
+  let newImage = null;
+
+  if (image && typeof image === 'object' && 'url' in image) {
```
좀더 단순하게 해도 될 거 같아요
if (image?.url) {

crongro reviewed (commented):
node에서도 이제 esm방식으로 개발하면 좋아요.
